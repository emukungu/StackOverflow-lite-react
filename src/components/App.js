import React from "react";
import { Route, Switch } from "react-router-dom";

import Signup from "../containers/signup";
const App = () => {
    return (
        <div>
            <Switch>
              <Route path = "/" component = { Signup }/>
            </Switch>
        </div>
    );
};

export default App;
