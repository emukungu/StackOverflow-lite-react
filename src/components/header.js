import React, { Component } from "react";
import { NavLink } from "react-router-dom";

export const logoutLink = (event) => {
  localStorage.removeItem("token");
};
export const HomeLink = () => {
  return (
    <NavLink to="/questions" className="home">
      Home
    </NavLink>
  );
};
export const HomeHeader = () => {
  return (
    <nav className=" navbar">
      <HomeLink />
      <NavLink to={"/login"} className="signupLinks login-link">
        Login
      </NavLink>

      <NavLink to={"/signup"} className="signupLinks">
        Sign Up
      </NavLink>
    </nav>
  );
};

export const LoggedInHeader = () => {
  return (
    <nav className=" navbar">
      <HomeLink />
      <NavLink
        to="/questions"
        onClick={logoutLink}
        className="signupLinks login-link">
        Logout
      </NavLink>
    </nav>
  );
};
export class Header extends Component {
  render() {
    const token = localStorage.getItem("token");
    if (token) {
      return <LoggedInHeader />;
    } else {
      return <HomeHeader />;
    }
  }
}
export default Header;
