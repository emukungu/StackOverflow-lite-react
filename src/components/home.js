import React, { Component } from "react";

export const Home = () => {
  return (
    <div className="container">
      <div className="row centered ">
        <div className=" col-xs-6 col-md-12  home-page">
          <h1>A PLACE FOR ALL YOUR QUESTIONS AND ANSWERS</h1>
          <p className="slogan"> Welcome to our community</p>
        </div>
      </div>
    </div>
  );
}

export default Home;
