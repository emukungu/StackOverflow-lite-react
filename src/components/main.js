import React from "react";
import { Route, Switch } from "react-router-dom";

import Signup from "../containers/signup";
import Login from "../containers/login";

import AllQuestions from "../containers/allQuestions";
import SpecificQuestion from "../containers/specificQn";
import PostQnContainer  from "../containers/postQuestion";
import Home  from "../components/home";
const Main = () => (
  <Switch>
    <Route exact path="/" component={Home} />
    <Route path="/signup" component={Signup} />
    <Route path="/login" component={Login} />
    <Route path="/postQuestion" component={PostQnContainer} />
    <Route exact path="/questions" component={AllQuestions} />
    <Route path="/questions/:id" component={SpecificQuestion} />
  </Switch>
);

export default Main;
