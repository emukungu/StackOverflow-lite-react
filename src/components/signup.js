import React from "react";

export const Input = props => {
  const { handleChange, type, id, placeholder, name, innerHTML , htmlFor} = props;
  return (
    <div className="form-group">
      <label htmlFor={htmlFor}>{innerHTML}</label>
      <div className="md-form">
        <input
          type={type}
          id={id}
          placeholder={placeholder}
          className="form-control"
          name={name}
          onChange={handleChange}
          required
        />
      </div>
    </div>
  );
};
export const SignupComponent = props => {
  const { handleSubmit, handleChange } = props;
  return (
    <div className="container">
      <div className="row">
        <div className=" col-xs-12 col-md-6">
          <h1>StackOverflow-Lite</h1>
          <p>A platform where people can ask questions and provide answers.</p>
        </div>
        <div className=" col-xs-12 col-md-6 center-signup">
          <h2>SIGN UP</h2>
          <form onSubmit={handleSubmit}>
            <Input htmlFor={"Username"}
                  innerHTML={"Username"}
                  name={"username"}
                  id={"username"}
                  handleChange={handleChange}
                  type={"text"}
                  placeholder={"Username"}/>
            <Input htmlFor={"Email"}
                  innerHTML={"Email"}
                  id={"email"}
                  name={"email"}
                  handleChange={handleChange}
                  type={"email"}
                  placeholder={"Email"}/>
            <Input htmlFor={"Password"}
                  innerHTML={"Password"}
                  id={"password"}
                  name={"password"}
                  handleChange={handleChange}
                  type={"password"}
                  placeholder={"Password"}/>
            <button type="submit" className="btn btn-default"> Submit </button>
          </form>
        </div>
      </div>
    </div>
  );
};

export default SignupComponent;
