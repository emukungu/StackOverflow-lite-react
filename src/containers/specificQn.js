import React, { Component } from "react";
import { connect } from "react-redux";

import specificQuestionAction from "../actions/specificQuestion";

export class SpecificQuestion extends Component {
  componentDidMount() {
    const { match, specificQn } = this.props;
    // const { id } = match.params;
    specificQn(match.params.id);
  }
  render() {
    const { question } = this.props;
    return (
      <div className="container ">
        <div className="row question-layout center-singleqn">
          <div className=" col-xs-12 col-md-12 qn-title">{question.title}</div>
          <div className=" col-xs-12 col-md-12 ">{question.description}</div>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = ({ allQnReducer }) => ({
  question: allQnReducer.question
});

export const mapDispatchToProps = dispatch => ({
  specificQn: id => dispatch(specificQuestionAction(id))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SpecificQuestion);
