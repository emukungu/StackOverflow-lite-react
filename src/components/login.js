import React from "react";
import { Input } from "./signup";

export const LoginComponent = props => {
  const { handleSubmit, handleChange } = props;
  return (
    <div className="container">
      <div className="row">
        <div className=" col-xs-12 col-md-6">
          <h1>StackOverflow-Lite</h1><p>A platform where people can ask questions and provide answers.</p>
        </div>
        <div className=" col-xs-12 col-md-6 center-signup"><h2>LOGIN</h2>
          <form onSubmit={handleSubmit} className="form-boundary">
            <Input htmlFor={"Username"} innerHTML={"Username"}
                name={"username"} id={"username"}
                handleChange={handleChange}
                type={"text"}
                placeholder={"Username"}/>
            <Input htmlFor={"Password"} innerHTML={"Password"}
                id={"password"} name={"password"}
                handleChange={handleChange}
                type={"password"}
                placeholder={"Password"} />
            <button type="submit" className="btn btn-default">Submit</button>
          </form>
        </div>
      </div>
    </div>
  );};

export default LoginComponent;
