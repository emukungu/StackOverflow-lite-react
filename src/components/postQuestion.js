import React from "react";
import { Input } from "./signup";

export const PostQn = props => {
  const { handleSubmit, handleChange } = props;
  return <div className="container">
      <div className="row">
        <div className=" col-xs-12 col-md-6 center-login">
          <h2>ADD A QUESTION</h2>
          <form onSubmit={handleSubmit} className="form-boundary">
            <Input htmlFor={"title"}
                innerHTML={"Title"}
                name={"title"}
                id={"title"}
                handleChange={handleChange}
                type={"text"}/>
            <div>
              <label htmlFor="description">Description</label>
              <textarea id="desc"
                  className="form-control post-qn"
                  name="desc"
                  onChange={handleChange}/>
            </div>
            <button type="submit" className="btn btn-default">Submit</button>
          </form>
        </div>
      </div>
    </div>
};

export default PostQn;
