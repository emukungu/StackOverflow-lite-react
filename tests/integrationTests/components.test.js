import React from "react";
import { shallow } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import App from "../../src/components/App";
import SignupComponent from "../../src/components/signup";
import LoginComponent from "../../src/components/login";
import { Header, HomeLink, HomeHeader, LoggedInHeader, logoutLink } from "../../src/components/header";
import Main from "../../src/components/main";
import { SpecificQuestion } from "../../src/containers/specificQn";
import Home from "../../src/components/home";

describe("<App />", () => {
  it("test App component", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
  it("tests the signup snapshot", () => {
    const component = shallow(<SignupComponent />);
    expect(component).toMatchSnapshot();
  });

  it("tests the login snapshot", () => {
    const component = shallow(<LoginComponent />);
    expect(component).toMatchSnapshot();
  });

  it("tests the header snapshot", () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
  it("tests the homeheader snapshot", () => {
    const component = shallow(<HomeHeader />);
    expect(component).toMatchSnapshot();
  });
  it("tests the LoggedInHeader snapshot", () => {
    const component = shallow(<LoggedInHeader />);
    expect(component).toMatchSnapshot();
  });

  it("tests the Main snapshot", () => {
    const component = shallow(<Main />);
    expect(component).toMatchSnapshot();
  });

  it("tests the Home snapshot", () => {
    const component = shallow(<Home />);
    expect(component).toMatchSnapshot();
  });
  it("logout successful", () => {
    logoutLink()
  });
  it("renders the component", () => {
    const question = { title: "worms", description: "lukewarm", qn_id: 2 };
    const component = shallow(
      <SpecificQuestion
        question={question}
        specificQn={jest.fn()}
        match={{ params: { id: 1 } }}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
