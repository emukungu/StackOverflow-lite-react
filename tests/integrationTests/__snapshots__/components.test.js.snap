// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> test App component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Switch>
        <Route
          component={[Function]}
          path="/"
        />
      </Switch>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Route
          component={[Function]}
          path="/"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
            component={[Function]}
            path="/"
          />
        </Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Route
            component={[Function]}
            path="/"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<App /> tests the signup snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignupComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="row"
      >
        <div
          className=" col-xs-12 col-md-6"
        >
          <h1>
            StackOverflow-Lite
          </h1>
          <p>
            A platform where people can ask questions and provide answers.
          </p>
        </div>
        <div
          className=" col-xs-12 col-md-6"
        >
          <form>
            <Input
              htmlFor="Username"
              id="username"
              innerHtml="Username"
              name="username"
              placeholder="Username"
              type="text"
            />
            <Input
              htmlFor="Email"
              id="email"
              innerHtml="Email"
              name="email"
              placeholder="Email"
              type="email"
            />
            <Input
              htmlFor="Password"
              id="password"
              innerHtml="Password"
              name="password"
              placeholder="Password"
              type="password"
            />
            <button
              className="btn btn-default"
              type="submit"
            >
               Submit 
            </button>
          </form>
        </div>
      </div>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className=" col-xs-12 col-md-6"
          >
            <h1>
              StackOverflow-Lite
            </h1>
            <p>
              A platform where people can ask questions and provide answers.
            </p>
          </div>,
          <div
            className=" col-xs-12 col-md-6"
          >
            <form>
              <Input
                htmlFor="Username"
                id="username"
                innerHtml="Username"
                name="username"
                placeholder="Username"
                type="text"
              />
              <Input
                htmlFor="Email"
                id="email"
                innerHtml="Email"
                name="email"
                placeholder="Email"
                type="email"
              />
              <Input
                htmlFor="Password"
                id="password"
                innerHtml="Password"
                name="password"
                placeholder="Password"
                type="password"
              />
              <button
                className="btn btn-default"
                type="submit"
              >
                 Submit 
              </button>
            </form>
          </div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                StackOverflow-Lite
              </h1>,
              <p>
                A platform where people can ask questions and provide answers.
              </p>,
            ],
            "className": " col-xs-12 col-md-6",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "StackOverflow-Lite",
              },
              "ref": null,
              "rendered": "StackOverflow-Lite",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "A platform where people can ask questions and provide answers.",
              },
              "ref": null,
              "rendered": "A platform where people can ask questions and provide answers.",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form>
              <Input
                htmlFor="Username"
                id="username"
                innerHtml="Username"
                name="username"
                placeholder="Username"
                type="text"
              />
              <Input
                htmlFor="Email"
                id="email"
                innerHtml="Email"
                name="email"
                placeholder="Email"
                type="email"
              />
              <Input
                htmlFor="Password"
                id="password"
                innerHtml="Password"
                name="password"
                placeholder="Password"
                type="password"
              />
              <button
                className="btn btn-default"
                type="submit"
              >
                 Submit 
              </button>
            </form>,
            "className": " col-xs-12 col-md-6",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Input
                  htmlFor="Username"
                  id="username"
                  innerHtml="Username"
                  name="username"
                  placeholder="Username"
                  type="text"
                />,
                <Input
                  htmlFor="Email"
                  id="email"
                  innerHtml="Email"
                  name="email"
                  placeholder="Email"
                  type="email"
                />,
                <Input
                  htmlFor="Password"
                  id="password"
                  innerHtml="Password"
                  name="password"
                  placeholder="Password"
                  type="password"
                />,
                <button
                  className="btn btn-default"
                  type="submit"
                >
                   Submit 
                </button>,
              ],
              "onSubmit": undefined,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleChange": undefined,
                  "htmlFor": "Username",
                  "id": "username",
                  "innerHtml": "Username",
                  "name": "username",
                  "placeholder": "Username",
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleChange": undefined,
                  "htmlFor": "Email",
                  "id": "email",
                  "innerHtml": "Email",
                  "name": "email",
                  "placeholder": "Email",
                  "type": "email",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleChange": undefined,
                  "htmlFor": "Password",
                  "id": "password",
                  "innerHtml": "Password",
                  "name": "password",
                  "placeholder": "Password",
                  "type": "password",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " Submit ",
                  "className": "btn btn-default",
                  "type": "submit",
                },
                "ref": null,
                "rendered": " Submit ",
                "type": "button",
              },
            ],
            "type": "form",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="row"
        >
          <div
            className=" col-xs-12 col-md-6"
          >
            <h1>
              StackOverflow-Lite
            </h1>
            <p>
              A platform where people can ask questions and provide answers.
            </p>
          </div>
          <div
            className=" col-xs-12 col-md-6"
          >
            <form>
              <Input
                htmlFor="Username"
                id="username"
                innerHtml="Username"
                name="username"
                placeholder="Username"
                type="text"
              />
              <Input
                htmlFor="Email"
                id="email"
                innerHtml="Email"
                name="email"
                placeholder="Email"
                type="email"
              />
              <Input
                htmlFor="Password"
                id="password"
                innerHtml="Password"
                name="password"
                placeholder="Password"
                type="password"
              />
              <button
                className="btn btn-default"
                type="submit"
              >
                 Submit 
              </button>
            </form>
          </div>
        </div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className=" col-xs-12 col-md-6"
            >
              <h1>
                StackOverflow-Lite
              </h1>
              <p>
                A platform where people can ask questions and provide answers.
              </p>
            </div>,
            <div
              className=" col-xs-12 col-md-6"
            >
              <form>
                <Input
                  htmlFor="Username"
                  id="username"
                  innerHtml="Username"
                  name="username"
                  placeholder="Username"
                  type="text"
                />
                <Input
                  htmlFor="Email"
                  id="email"
                  innerHtml="Email"
                  name="email"
                  placeholder="Email"
                  type="email"
                />
                <Input
                  htmlFor="Password"
                  id="password"
                  innerHtml="Password"
                  name="password"
                  placeholder="Password"
                  type="password"
                />
                <button
                  className="btn btn-default"
                  type="submit"
                >
                   Submit 
                </button>
              </form>
            </div>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1>
                  StackOverflow-Lite
                </h1>,
                <p>
                  A platform where people can ask questions and provide answers.
                </p>,
              ],
              "className": " col-xs-12 col-md-6",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "StackOverflow-Lite",
                },
                "ref": null,
                "rendered": "StackOverflow-Lite",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "A platform where people can ask questions and provide answers.",
                },
                "ref": null,
                "rendered": "A platform where people can ask questions and provide answers.",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <form>
                <Input
                  htmlFor="Username"
                  id="username"
                  innerHtml="Username"
                  name="username"
                  placeholder="Username"
                  type="text"
                />
                <Input
                  htmlFor="Email"
                  id="email"
                  innerHtml="Email"
                  name="email"
                  placeholder="Email"
                  type="email"
                />
                <Input
                  htmlFor="Password"
                  id="password"
                  innerHtml="Password"
                  name="password"
                  placeholder="Password"
                  type="password"
                />
                <button
                  className="btn btn-default"
                  type="submit"
                >
                   Submit 
                </button>
              </form>,
              "className": " col-xs-12 col-md-6",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Input
                    htmlFor="Username"
                    id="username"
                    innerHtml="Username"
                    name="username"
                    placeholder="Username"
                    type="text"
                  />,
                  <Input
                    htmlFor="Email"
                    id="email"
                    innerHtml="Email"
                    name="email"
                    placeholder="Email"
                    type="email"
                  />,
                  <Input
                    htmlFor="Password"
                    id="password"
                    innerHtml="Password"
                    name="password"
                    placeholder="Password"
                    type="password"
                  />,
                  <button
                    className="btn btn-default"
                    type="submit"
                  >
                     Submit 
                  </button>,
                ],
                "onSubmit": undefined,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "handleChange": undefined,
                    "htmlFor": "Username",
                    "id": "username",
                    "innerHtml": "Username",
                    "name": "username",
                    "placeholder": "Username",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "handleChange": undefined,
                    "htmlFor": "Email",
                    "id": "email",
                    "innerHtml": "Email",
                    "name": "email",
                    "placeholder": "Email",
                    "type": "email",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "handleChange": undefined,
                    "htmlFor": "Password",
                    "id": "password",
                    "innerHtml": "Password",
                    "name": "password",
                    "placeholder": "Password",
                    "type": "password",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Submit ",
                    "className": "btn btn-default",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": " Submit ",
                  "type": "button",
                },
              ],
              "type": "form",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
